workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs

```diff
@@ -1,54 +1,71 @@
+using System.Collections.Generic;
 using Improbable.Gdk.Core;
-using Improbable.Gdk.ReactiveComponents;
-using Unity.Collections;
+using Improbable.Worker.CInterop;
 using Unity.Entities;
 using UnityEngine;
 
 namespace Improbable.Gdk.Health
 {
     [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
     [UpdateAfter(typeof(ServerHealthModifierSystem))]
+    [AlwaysUpdateSystem]
     public class HealthRegenSystem : ComponentSystem
     {
-        public struct EntitiesNeedingRegenData
+        private WorkerSystem workerSystem;
+        private ComponentUpdateSystem componentUpdateSystem;
+        private CommandSystem commandSystem;
+
+        private ComponentGroup initGroup;
+        private ComponentGroup regenGroup;
+
+        private HashSet<EntityId> recentlyDamagedCache = new HashSet<EntityId>();
+
+        protected override void OnCreateManager()
         {
-            public readonly int Length;
-            [ReadOnly] public EntityArray Entities;
-            [ReadOnly] public ComponentDataArray<HealthRegenComponent.Component> HealthRegenComponents;
-            [ReadOnly] public SubtractiveComponent<HealthRegenData> DenotesMissingData;
-            [ReadOnly] public ComponentDataArray<Authoritative<HealthComponent.Component>> DenotesAuthority;
+            base.OnCreateManager();
+
+            workerSystem = World.GetExistingManager<WorkerSystem>();
+            componentUpdateSystem = World.GetExistingManager<ComponentUpdateSystem>();
+            commandSystem = World.GetExistingManager<CommandSystem>();
+
+            initGroup = GetComponentGroup(ComponentType.ReadOnly<HealthRegenComponent.Component>(),
+                ComponentType.Subtractive<HealthRegenData>(),
+                ComponentType.ReadOnly<HealthComponent.ComponentAuthority>()
+            );
+            initGroup.SetFilter(HealthComponent.ComponentAuthority.Authoritative);
+
+            regenGroup = GetComponentGroup(
+                ComponentType.Create<HealthRegenComponent.Component>(),
+                ComponentType.Create<HealthRegenData>(), ComponentType.ReadOnly<HealthComponent.Component>(),
```

Comment:
(Nit) Lets put one on each line otherwise your eyes skip over the second one when its the odd man out.

Reply:
Oops, forgot that here, will do